controllers/activemqartemis_reconciler.go

```diff
@@ -1222,8 +1295,10 @@ func generateAcceptorSSLOptionalArguments(acceptor brokerv1beta1.AcceptorType) s
 		sslOptionalArguments = sslOptionalArguments + ";" + "keyStoreProvider=" + acceptor.KeyStoreProvider
 	}
 
-	if acceptor.TrustStoreType != "" {
-		sslOptionalArguments = sslOptionalArguments + ";" + "trustStoreType=" + acceptor.TrustStoreType
+	if acceptor.KeyStoreType != "" {
+		if sslArgs != nil && len(sslArgs.PemCfgs) > 2 {
```

Comment:
should we respect the keyStoreType always? and then do further work when it is PEM? That leaves a user open to mal configuration, but I think it is more odd if it is ignored. When the secret is provided by the user, then it could well be of the correct type/format. I guess it does not make sense to preclude that if possible.

I note that we hide the PEMCFG, we know that there will be two sources of info in PEM format, and make use of PEMCFG when necessary. I don't know if we loose anything by keeping that hidden. 
Interested in knowing your thoughts.

Reply:
I'll make it optional and auto detect the type for PEM/PEMCFG if not specified. And add some checks if user specifies the wrong type.